taskboard/types.py

```diff
@@ -0,0 +1,13 @@
+from dataclasses import dataclass
+from typing import Optional
+
+@dataclass
+class Task:
+    id: str
+    title: str
+    status: Optional[str] = None
+
+@dataclass
+class TaskList:
```

Comment:
what is the difference between this and a Task?

Reply:
TaskList is a container for tasks. For ex: There can be a TaskList for each course (OSPSD, OS, NetSec, DSA). 
Task is the action.

Reply:
how is it a container when it only has `id` and `name` fields -- it feels like 1 task?

Reply:
it's not a hierarchical container that it stores tasks inside it, rather the relationship between a task and tasklist will be defined in the storage (like the database). 
example storage:     `{"id": "1", "title": "a", "list_id": "ospsd"}`
implementation of get_tasks will query the storage engine to  find all tasks with a matching list_id. 
we could add `list_id` field to `Task` dataclass, but its additional metadata and not strictly necessary.

Reply:
in what context would the user care for this?
the database schema (if that) seems like an implementation detail.
the user would want to get all the tasks in a list (which should return `list[Task]`) or the list of a task..
if a task can be in only one list, then a `list_id` or `parent_list` would be a reasonable thing to have on a `Task`

Reply:
as written this is confusing at best and unclear how to use

Reply:
changed it so that Task dataclass has a list_id.